src/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/SignedXml.cs

```diff
@@ -361,12 +361,12 @@ public bool CheckSignature(X509Certificate2 certificate, bool verifySignatureOnl
 
                 if (!chainVerified)
                 {
-                    SignedXmlDebugLog.LogVerificationFailure(this, SecurityResources.GetResourceString("Log_VerificationFailed_X509Chain"));
+                    SignedXmlDebugLog.LogVerificationFailure(this, SR.GetResourceString("Log_VerificationFailed_X509Chain"));
                     return false;
                 }
             }
 
-            if (!CheckSignature(certificate.GetAnyPublicKey()))
+            if (!CheckSignature(certificate.PublicKey.Key))
```

Comment:
Not quite the same.  `GetRSAPublicKey() ?? GetDSAPublicKey()` is more apt.  Though, now that I'm here I want to say "but it should be in a using statement".